docs/source/point-evaluation.rst

```diff
@@ -104,6 +106,186 @@ in parallel. There is no special API, but there are some restrictions:
 * Each process will get the same values.
 
 
+Firedrake API 2: Interpolation onto a vertex-only mesh
+------------------------------------------------------
+
+Firedrake's second API for evaluating functions at arbitrary points,
+interpolation onto a :func:`~.VertexOnlyMesh`, is designed for evaluating a
+function at many points and for creating expressions which contain point
+evaluations. It has been designed from the ground up to be entirely parallel
+compatible. Whilst :meth:`~.Function.at` produces a list of values, a
```

Comment:
It would be nice to see a bit more detail on how to use it in parallel rather than referring to the docstring as you do later on. Is the warning in the docstring

> This means that if rank A has ``vertexcoords`` {X} that are not found in the mesh cells owned by rank A but are found in the mesh cells owned by rank B, **and rank B has not been supplied with those**, then they will be moved to rank B.

stil valid? In my interpretation reading the code - but correct me if I'm wrong - with `redundant=False` every process specifies unique points which get concatenated with an algatherv, and then every local process keeps its locally visible ones with some voting algorithm to ensure each point gets assignted to only one process.

Regardless of how the points are specified (redundant or not), I'm a little unclear on how the interpolated data is meant to be used in parallel. The `.dat.data` corresponds to the interpolated data of the locally visible detectors, but as a user I have no idea which ones they are and how they are related to the points I've specified and in which order.

Reply:
> stil valid? In my interpretation reading the code - but correct me if I'm wrong - with redundant=False every process specifies unique points which get concatenated with an algatherv, and then every local process keeps its locally visible ones with some voting algorithm to ensure each point gets assignted to only one process.

It is still valid, in fact it's a new feature as of #2773! The point is that you can specify a point on one rank and, if it's not on that rank but is on another, it'll be moved to that rank. See issue #2178 where this was first raised.

>Regardless of how the points are specified (redundant or not), I'm a little unclear on how the interpolated data is meant to be used in parallel. The .dat.data corresponds to the interpolated data of the locally visible detectors, but as a user I have no idea which ones they are and how they are related to the points I've specified and in which order.

I don't quite understand you here, what do you mean by detectors?

Reply:
Well you can get the point coordinates from the VoM. However I think this exposes a hole in the current implementation. 

Think about the use case of interpolating from one mesh to another, then the point cloud starts with one decomposition. It necessarily gets rearranged to another parallel decomposition in order to match the parent mesh and do interpolation. However, the user needs the points back in the original configuration to use the data. 

I think we need to store the original rank and local number of every point and preserve this as we transmit the points around (these could be fields on the DMSwarm). Then we could provide a method that gives you back the points in the original decomposition. I guess we have to think about what the type that that would return would be.

Reply:
Oh heh sorry, on the docstring: I read what it says here https://www.firedrakeproject.org/firedrake.html#firedrake.mesh.VertexOnlyMesh (which seems out-of-date): "then the vertexcoords {X} will be lost." but then I copied from the code, which is correct indeed.

That was exactly the application I was thinking of indeed.

Another application I can think of is interpolating onto an external grid, say a regular structured one. I guess in that situation you would typically use `redundant=True`, but you have the same issue that it'll be hard to piece the local interpolated values in each .dat.data back to a structured array. Is this something DMSWARM can do? We do give it the original global index of the points as provided.

Reply:
Yep looks like the website needs to be rebuilt @stephankramer . @dham I wonder if this moves us in a direction of needing VoMs which have no parent...

@stephankramer I suppose I could start saving the interpolated values as a field on the DMSWARM...

Reply:
@stephankramer Okay I've expanded the manual to include more information

Reply:
Also you made me realise there's a mistake in the VertexOnlyMesh API docs that I've now fixed 82f91141a1035e56941d1dbc774c6611742d2566